doc/reference/configuration/configuration_reference.rst

```diff
@@ -2877,6 +2878,120 @@ This section describes configuration parameters related to roles.
 
     **Since:** :doc:`3.0.0 </release/3.0.0>`.
 
+    Specify the roles of an instance.
+    To specify a role's configuration, use the :ref:`roles_cfg <configuration_reference_roles_cfg>` option.
+
+    There are two types of application roles:
+
+    -   Built-in roles. For example, the `CRUD <https://github.com/tarantool/crud>`__ module provides the ``roles.crud-storage`` and ``roles.crud-router`` roles that enable CRUD operations in a sharded cluster.
+    -   Custom roles. For example, you create a custom role to define a stored procedure or implement a supplementary service, such as an e-mail notifier or a replicator.
+
+    To create a custom role, you need to define the following methods and fields:
+
+    -   ``validate()``
+
+        Validate a role's configuration.
+        This method is called when the initial configuration is applied or the configuration is reloaded for the instance with this role.
+        ``validate()`` should throw an error if the validation fails.
```

Comment:
'When the initial configuration is applied' means the same what 'on instance startup' means, but the latter looks simpler.

First time I read it, my thought was about a first `require` of the module (because I thought about a role's configuration, not the whole configuration).